contracts/src/v0.6/Oracle.sol

```diff
@@ -27,6 +27,16 @@ contract Oracle is ChainlinkRequestInterface, OracleInterface, Ownable, LinkToke
   mapping(address => bool) private authorizedNodes;
   uint256 private withdrawableTokens = ONE_FOR_CONSISTENT_GAS_COST;
 
+  // Struct packed into 2 slots
+  // Cannot pack into 1 because we need payment to be > uint72 and expiration uint32
+  struct PackedOracleRequest {
+    address callbackAddress; // slot1: 160
+    bytes4 callbackFunctionId; // slot1: 160 + 32
+    uint128 payment; // slot2: 128
+    uint128 expiration; // slot2: 128 + 128
+  }
+  mapping(bytes32 => PackedOracleRequest) private s_packedRequests;
```

Comment:
Summary: Moving 4 callback vars to storage. Payment and expiration have been cast from uint256 for packing purposes. These are 4 of 6 initially part of fulfill callback. Other 2 (requestId, data) are not able to be moved out. RequestId is used to fetch storage data, data comes from off-chain.

Reply:
Thought: Can we dynamically fulfill based on chain market conditions? If calldata is more expensive, use storage, otherwise use callback.